gui.py

```diff
@@ -12,99 +14,169 @@ def __init__(self, parent=None):
+        self.main_layout.addWidget(self.game_path_info_label)
 
-        about_label = QtGui.QLabel()
-        main_layout.addWidget(api_token_label)
-        main_layout.addWidget(self.api_token_tb)
-        main_layout.addWidget(api_token_info_label)
+        self.main_layout.addWidget(craft_list_label)
+        self.main_layout.addWidget(self.craft_list_ql)
 
-        main_layout.addWidget(title_label)
-        main_layout.addWidget(self.title_tb)
-        main_layout.addWidget(title_info_label)
+        self.main_layout.addWidget(api_token_label)
+        self.main_layout.addWidget(self.api_token_tb)
+        self.main_layout.addWidget(api_token_info_label)
 
-        main_layout.addWidget(description_label)
-        main_layout.addWidget(self.description_tb)
-        main_layout.addWidget(description_info_label)
+        self.main_layout.addWidget(name_label)
+        self.main_layout.addWidget(self.name_tb)
 
-        main_layout.addWidget(tags_label)
-        main_layout.addWidget(self.tags_tb)
-        main_layout.addWidget(tags_info_label)
+        self.main_layout.addWidget(self.description_label)
+        self.main_layout.addWidget(self.description_tb)
 
-        main_layout.addWidget(craft_list_label)
-        main_layout.addWidget(self.craft_list_ql)
+        self.main_layout.addWidget(tags_label)
+        tag_h_layout = QtGui.QHBoxLayout()
+        tag_h_layout.addWidget(ksp_tag_label)
+        tag_h_layout.addWidget(self.tags_tb)
+        self.main_layout.addLayout(tag_h_layout)
 
-        main_layout.addWidget(self.status_info_label)
-        self.update_game_gui(self.game_dir)
+        self.main_layout.addWidget(self.status_info_label)
         self.craft_list_ql.currentRowChanged.connect(self.updateList)
-        main_layout.addWidget(self.upload_btn)
-        main_layout.addWidget(self.close_btn)
-        self.setLayout(main_layout)
+        self.main_layout.addWidget(self.upload_btn)
+        self.setLayout(self.main_layout)
+
+        # Build manager and set ui
+        self.manager = KSP2Skfb(self.game_dir, True)
+        self.update_game_ui(self.game_dir)
+        self.enable_skfb_ui(False)
+        QtCore.QObject.connect(self.manager.emitter, QtCore.SIGNAL('building(QString, int, int)'), self.update_upload_btn)
+
+        # If bad path at launch, pop up the filedialog to locate it
+        if not os.path.exists(self.game_dir):
+            QtGui.QMessageBox.warning(self, "Game directory not found", "Please set your game directory")
+            self.search_game_directory()
+
+    def get_user_data(self):
+        if os.path.exists('userdata.json'):
+            need_clear = False
+            with open('userdata.json', 'r') as f:
+                try:
+                    self.user_data = json.load(f)
+                    print(self.user_data)
+                except:
+                    need_clear = True
+                    print(os.getcwd())
+            if need_clear:
+                os.remove('userdata.json')
```

Comment:
why the need for an extra variable? the only code path that leads to this is when an exception is raised; move the `os.remove` in the exceptino block. The `print` statement is likely a debug stuff; if so remove it